### PR TITLE
Replaced removed NetworkInterfaces api

### PR DESCRIPTION
**Pull request by mchf**
See https://github.com/yast/yast-yast2/pull/789

### CODE REVIEW

File: src/modules/LanItems.rb [hunk 1]
```diff
@@ -2530,6 +2523,29 @@ def clear_set_hostname
...
+    # Raises an exception when type is incorrect.
+    #
+    # @param [String] device type
+    # @return [String] available device name
+    def new_type_device(type)
+      raise ArgumentError, "Valid device type expected" if type.nil? || type.empty?
+
+      known_devs = find_type_ifaces(type)
+
+      return type << "0" if known_devs.empty?
```
Comment (mvidner) [inline src/modules/LanItems.rb]: 1. The special case is unnecessary. Without it, the method produces the correct result, efficiently.
2. String#<< modifies the receiver. So this method mutates its argument, which happens to be `@type`!

File: src/modules/LanItems.rb [hunk 2]
```diff
@@ -2530,6 +2523,29 @@ def clear_set_hostname
...
+
+      candidates = Array(0..known_devs.size).map { |c| "#{type}#{c}" }
```
Comment (mvidner) [inline src/modules/LanItems.rb]: `(0..n)` is a Range which is Enumerable, so the Array call is unnecessary.

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -2530,6 +2523,29 @@ def clear_set_hostname
       ret
     end
 
+    # Returns unused name for device of given type
+    #
+    # When already having eht0, eth1, enp0s3 devices (eth type) and asks for new
```
[the remark this answers is not in the document]
Comment (mchf) [reply src/modules/LanItems.rb]: only an example of eth device with other then good old name

File: package/yast2-network.changes [hunk 4]
```diff
@@ -1,3 +1,11 @@
+-------------------------------------------------------------------
+Tue Sep 18 07:31:10 UTC 2018 - mfilka@suse.com
+
+- bnc#964856
+  - adapted to changes in NetworkInterfaces API
```
[the remark this answers is not in the document]
Comment (mchf) [reply package/yast2-network.changes]: nice tip, will try. thanks

File: src/modules/LanItems.rb [hunk 5]
```diff
@@ -2774,6 +2790,18 @@ def find_by_sysconfig
       GetDeviceNames(items)
     end
 
+    # Finds all items of given device type
+    #
+    # @param type [String] device type
+    # @return [Array] list of device names
```
[the remark this answers is not in the document]
Comment (mchf) [reply src/modules/LanItems.rb]: ok

**Pull request merged**